girder_worker/docker/tasks/__init__.py

```diff
@@ -359,6 +363,44 @@ def _cleanup_temp_volumes(self, temp_volumes, default_temp_volume):
             shutil.rmtree(v.host_path)
 
 
+#Class for SingularityTask similar to DockerTask
+class SingularityTask(Task):
+    def __call__(self, image,*args,container_args,bind_paths,**kwargs):
+        image = image or kwargs.pop('image',None)
+        container_args = container_args or kwargs.pop('container_args',[])
+        bind_paths = bind_paths or kwargs.pop('bind_paths',{})
+        temporary_directory = os.getenv('TEMPORARY_DIRECTORY','/tmp')
+        log_file_path = os.getenv('SINGULARITY_LOG_FILE','/log')
+        
+        #check if the user has provided the image of the plugin.
+        if not image:
+            image = "sarderlab/histo-cloud:latest"
+            # raise ValueError('Plugin Image required for Singularity')
+        
+        #Commnad to be called for executing Singularity Job
+        bind_paths[temporary_directory] = '/output'
+        super().__call__(*args,**kwargs)
+        
+        qos = kwargs.pop('qos', 'pinaki.sarder')
```

Comment:
I'd recommend having these from environment variables.  And, rather than construct the slurm script ourselves, we could use the `simple_slurm` package to do it (it does the same thing, but has been validated by others and escapes command line parameters appropriately).